cvxpy/tests/test_python_backends.py

```diff
@@ -67,28 +69,68 @@ def test_get_backend(self):
             CanonBackend.get_backend('notabackend')
 
 
-class TestScipyBackend:
+class TestBackends:
     # Not used explicitly in most test cases.
     # Some tests specify other values as needed within the test case.
     param_size_plus_one = 2
     id_to_col = {1: 0, 2: 2}
     param_to_size = {-1: 1, 3: 1}
     param_to_col = {3: 0, -1: 1}
     var_length = 4
+    const = "numpy"
 
     @pytest.fixture
-    def backend(self):
+    def scipy_backend(self):
         return ScipyCanonBackend(self.id_to_col, self.param_to_size, self.param_to_col,
                                  self.param_size_plus_one, self.var_length)
 
+    @pytest.fixture
+    def numpy_backend(self):
+        return NumpyCanonBackend(self.id_to_col, self.param_to_size, self.param_to_col,
+                                 self.param_size_plus_one, self.var_length)
+
+    @pytest.fixture(params=[const + "_backend"])
+    def backend(self, request):
+        return request.getfixturevalue(request.param)
+
+    @pytest.fixture()
+    def numpy_arg_view(self):
+        def numpy_view(param_size_plus_one=None, id_to_col=None,
+                       param_to_size=None, param_to_col=None,
+                       var_length=None):
+            return NumpyTensorView.get_empty_view(param_size_plus_one or self.param_size_plus_one,
+                                                  id_to_col or self.id_to_col,
+                                                  param_to_size or self.param_to_size,
+                                                  param_to_col or self.param_to_col,
+                                                  var_length or self.var_length)
+
+        return numpy_view
```

Comment:
This is a fixture that can receive arguments, but defaults to the empty_view with given (self.arguments).
Find the reference [here](https://stackoverflow.com/questions/44677426/can-i-pass-arguments-to-pytest-fixtures).

Reply:
Very nice!